### PR TITLE
Add shell.background-color.rgba to prefs

### PR DESCRIPTION
**Pull request by wusyong**
With [surfman#260](https://github.com/servo/surfman/pull/260), it's possible for all platforms to display transparent background now. This could let users to make more native-like interface as they wish. Right now, servo will still make viewport background [white](https://github.com/servo/servo/blob/master/components/compositing/compositor.rs#L1854). And I hope it can add a config to toggle this to transparent background.

Based on feedbacks on zulip, this PR adds `shell.background-color.rgba` to prefs with the resource file `prefs.json` set to `[1.0, 1.0, 1.0, 1.0]` by default.

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes doesn't fix any github issue.
- [x] These changes do not require tests because this is only a background for shell's viewport.

### CODE REVIEW

**Comment by mrobinson** [comment]
I really think should allow setting a background color (which can be transparent) as that is what all other WebView APIs. do.

**Comment by wusyong** [comment]
Alright, how about changing it to `background_color: { r: f64, g: f64, b: f64, a: f64}` (as I didn't see the option to set it as f32 in prefs).

**Comment by mrobinson** [comment]
> Alright, how about changing it to `background_color: { r: f64, g: f64, b: f64, a: f64}` (as I didn't see the option to set it as f32 in prefs).

That seems sensible. Thanks.

File: resources/prefs.json [hunk 1]
```diff
@@ -110,6 +110,10 @@
   "network.http-cache.disabled": false,
   "network.mime.sniff": false,
   "session-history.max-length": 20,
+  "shell.background-color.alpha": 1.0,
+  "shell.background-color.blue": 1.0,
+  "shell.background-color.green": 1.0,
+  "shell.background-color.red": 1.0,
```
[the remark this answers is not in the document]
Comment (wusyong) [reply resources/prefs.json:116]: It'll need to add new variant like serde's `Value::Array` to `PrefValue`. I can try to add this.

[the remark this answers is not in the document]
Comment (wusyong) [reply resources/prefs.json]: @mrobinson I added new variant `PrefValue` and make sure other crates can handle it well.
It probably could add a macro to handle `From` trait between array and this vraint.
But I suppose it's better to wait until there's more configs that require array.
So we can implement more ergonomic one.

**Review by mrobinson — APPROVED** [review]

**Comment by mrobinson** [comment]
Nice.

**Pull request merged**